### DISCUSSION TITLE
Leniency for generic types with [StructLayout(LayoutKind.Explicit)]

### DISCUSSION THREAD

**Discussion by tagcode** [dotnet/csharplang#8443] [General]
Recreated at https://github.com/dotnet/runtime/discussions/108063

**Comment by HaloFour** [comment]
I think this is a [runtime](https://github.com/dotnet/runtime) request.  The C# language itself doesn't interpret or enforce those attributes, which is why you get an exception at runtime and not during compile time.

**Reply by tagcode** [reply]
Thanks for link. I'll post there, and remove this one.
